apps/reference/docs/glossary.mdx

```diff
@@ -8,18 +8,194 @@ Definitions for terminology and acronyms used in the Supabase documentation.
 
 ## Auth
 
-### Access token
+### Tokens
 
-### Multi-factor authentication (MFA) {#multi-factor-authentication}
+A token is a string of characters given to a client (web browser, server).
 
-### One-time password (OTP) {#one-time-password}
+#### Access token
 
-### Provider token
+An access token is a short-lived (usually no more than 1 hour) token that
+authorizes a client to access resources on a server. It comes in the form of a
+JWT (JSON Web Token).
 
-### Provider refresh token
+#### Refresh token
 
-### Refresh token
+A refresh token is a long-lived (in most cases with an indefinite lifetime)
+token that is meant to be stored and exchanged for a new refresh and access
+tokens only once. Once a refresh token is exchanged it becomes invalid, and
+can't be exchanged again. In practice, though, a refresh token can be exchanged
+multiple times but in a short time window.
 
+#### Provider tokens
+
+A provider token is a long-lived token issued by a third-party identity
+provider. These are issued by social login services (Google, Twitter, Apple,
+Microsoft, ...) and uniquely identify a user on those platforms.
+
+### Authentication
+
+Authentication (often abbreviated authn.) is the process of verifying the
+identity of a user. Verification of the identity of a user can happen in
+multiple ways:
+
+1. Asking users for something they know. For example: password, passphrase.
+2. Checking that users have access to something they own. For example: an email
+address, a phone number, a hardware key, recovery codes.
+3. Confirming that users have some biological features. For example: a
+fingerprint, a certain facial structure, an iris print.
+
+### Authorization
+
+Authorization (often abbreviated authz.) is the process of verifying if a
+certain identity is allowed to access resources. Authorization often occurs by
+verifying an access token.
+
+### Session
```

Comment:
should we also mention here that refreshing the access token will lead to continuation of the existing session?